network/common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java

```diff
@@ -50,6 +55,8 @@
 
   private final Map<Long, RpcResponseCallback> outstandingRpcs;
 
+  private final Queue<StreamCallback> streamCallbacks;
```

Comment:
Yes, it can be a queue. The `StreamRequest` RPC being added is responded in the order they are sent (the netty read path is single threaded, and the handler for this message responds to it immediately, instead of dispatching to some thread pool).